debezium-core/src/main/java/io/debezium/pipeline/notification/IncrementalSnapshotNotificationService.java

```diff
@@ -131,8 +132,11 @@ public <T extends DataCollectionId> void notifyInProgress(IncrementalSnapshotCon
                 Map.of(
                         DATA_COLLECTIONS, dataCollections,
                         CURRENT_COLLECTION_IN_PROGRESS, incrementalSnapshotContext.currentDataCollectionId().getId().identifier(),
-                        MAXIMUM_KEY, Objects.toString(incrementalSnapshotContext.maximumKey().orElse(new Object[0])[0], "<null>"),
-                        LAST_PROCESSED_KEY, Objects.toString(incrementalSnapshotContext.chunkEndPosititon()[0], "<null>")),
+                        MAXIMUM_KEY, incrementalSnapshotContext.maximumKey().isPresent() ? Arrays.stream(incrementalSnapshotContext.maximumKey().get())
```

Comment:
@vjuranek Can you use `map`/`orElse` combination instead of ternary operator to have the code fully functional style instead of mixed?

Reply:
thanks, good point